PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:5 + add:0, change matches:
-    _inherit = ['hr.contract']
-    _inherit = ['hr.leave.type']
-    _inherit = ['hr.leave']
-    _inherit = ['hr.work.entry']
-    _inherit = ['hr.work.entry.type']

Total Changes: 10

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/hr_work_entry_holidays/models/hr_contract.py
+++ b/addons/hr_work_entry_holidays/models/hr_contract.py
@@ -8,7 +8,7 @@ from odoo.osv.expression import OR
 
 
 class HrContract(models.Model):
-    _inherit = ['hr.contract']
+    _inherit = 'hr.contract'
     _description = 'Employee Contract'
 
     # override to add work_entry_type from leave

--- a/addons/hr_work_entry_holidays/models/hr_leave.py
+++ b/addons/hr_work_entry_holidays/models/hr_leave.py
@@ -12,13 +12,13 @@ from odoo.tools import format_date
 
 
 class HrLeaveType(models.Model):
-    _inherit = ['hr.leave.type']
+    _inherit = 'hr.leave.type'
 
     work_entry_type_id = fields.Many2one('hr.work.entry.type', string='Work Entry Type')
 
 
 class HrLeave(models.Model):
-    _inherit = ['hr.leave']
+    _inherit = 'hr.leave'
 
     def _prepare_resource_leave_vals(self):
         vals = super(HrLeave, self)._prepare_resource_leave_vals()

--- a/addons/hr_work_entry_holidays/models/hr_work_entry.py
+++ b/addons/hr_work_entry_holidays/models/hr_work_entry.py
@@ -7,7 +7,7 @@ from odoo import api, fields, models
 
 
 class HrWorkEntry(models.Model):
-    _inherit = ['hr.work.entry']
+    _inherit = 'hr.work.entry'
 
     leave_id = fields.Many2one('hr.leave', string='Time Off')
     leave_state = fields.Selection(related='leave_id.state')
@@ -101,7 +101,7 @@ class HrWorkEntry(models.Model):
 
 
 class HrWorkEntryType(models.Model):
-    _inherit = ['hr.work.entry.type']
+    _inherit = 'hr.work.entry.type'
     _description = 'HR Work Entry Type'
 
     leave_type_ids = fields.One2many(
